waku/waku_archive/driver/builder.nim

```diff
@@ -107,6 +107,19 @@ proc new*(T: type ArchiveDriver,
         if migrateRes.isErr():
           return err("ArchiveDriver build failed in migration: " & $migrateRes.error)
 
+      ## This should be started once we make sure the 'messages' table exists
+      ## Hence, this should be run after the migration is completed.
+      asyncSpawn driver.startPartitionFactory(onFatalErrorAction)
```

Comment:
Is this just an async proc start isn't? 
How do you see the regular partition management blocking overhead on general message processing?
Would it be possible to separate this management into a complete new thread instead? I mean to consider this option.

Reply:
> Is this just an async proc start isn't? How do you see the regular partition management blocking overhead on general message processing? Would it be possible to separate this management into a complete new thread instead? I mean to consider this option.

This doesn't represent a big issue because the partitons info is stored in memory and the interaction with database to create a new partition will happen once per hour with that configuration.